include/net/net_pkt.h

```diff
@@ -396,6 +414,19 @@ static inline void net_pkt_set_ipv4_ttl(struct net_pkt *pkt,
 	ARG_UNUSED(pkt);
 	ARG_UNUSED(ttl);
 }
+
+static inline u8_t net_pkt_ipv4_opts_len(struct net_pkt *pkt)
+{
+	ARG_UNUSED(pkt);
+	return 0;
+}
+
+static inline void net_pkt_set_ipv4_opts_len(struct net_pkt *pkt,
+					     u8_t opts_len)
+{
+	ARG_UNUSED(pkt);
+	ARG_UNUSED(opts_len);
+}
```

Comment:
could you add a ip version agnostic net_pkt_ip_opts_len() which would return ipv4_opts_len if CONFIG_NET_IPV4 is the only enabled or ipv6_ext_len if CONFIG_NET_IPV6 is enabled?

It's ok to return ipv6_ext_len even if the pkt is ipv4. When allocated, the net_pkt is memset to 0.


See below why this would be needed.